treee111/wahooMapsCreator
# [FEAT] install: Download Osmosis and Osmfilter on Windows during first run instead of shipping with the tool

## This PR…

- downloads OSM-tools (Osmosis, Osmfilter) on Windows if it is not already there
- brings down to-be-published size of wahooMapsCreator down from 29MB to 1,5MB

## Considerations and implementations

Reasons for not shipping Osmosis via PyPI:
- only relevant for Windows
- hugh part of the published tool in percentage
- only needed "once", does not change per new version (actually)

Reason for not shipping Osmconvert:
- the "[official](http://m.m.i24.cc/osmconvert64.exe)" is not producting output on Windows and might not be working with very big files - see [Limitations](https://wiki.openstreetmap.org/wiki/Osmconvert#Windows)
- hence I continue using `osmconvert64-0.8.8p.exe` and the non 64bit version from within the python module

without Osmosis and Osmfilter: wahoomc-3.2.0a3 / branch dl-tooling-win:
<img width="510" alt="grafik" src="https://user-images.githubusercontent.com/53038537/204459837-24aeb670-68ef-41b8-a362-7efaf0142a64.png">

with Osmosis and Osmfilter: wahoomc-3.2.0a4 / branch develop:
<img width="499" alt="grafik" src="https://user-images.githubusercontent.com/53038537/204460247-0514c8e1-09c2-410c-a3d6-a210dd7e767d.png">



## How to test

1. install version `wahoomc-3.2.0a3` or checkout this branch
2. check this folder in your user directory `wahooMapsCreatorData/_download` for a `tooling_win` folder
- in both OS, this folder should be not there
4. run `python -m wahoomc cli -co malta` or `python -m wahoomc -h`
2. check this folder in your user directory `wahooMapsCreatorData/_download` for a `tooling_win` folder
- in Windows there should be a `Osmosis` directory and a `osmfilter.exe` file
![grafik](https://user-images.githubusercontent.com/53038537/204565582-88bebe36-9e4d-4212-9ad4-b3f8c917e6e7.png)

- in macOS this folder should be not there


## Pull Request Checklist
    + Especially the [Pull-Requests](https://github.com/treee111/wahooMapsCreator/blob/develop/.github/CONTRIBUTING.md#Pull-Requests) section
